account_invoice_report_grouped_by_picking/models/account_move.py

```diff
@@ -108,6 +108,10 @@ def lines_grouped_by_picking(self):
                         )
                         picking_dict.setdefault(key, 0)
                         qty = so_line.product_uom_qty
+                        if remaining_qty < so_line.product_uom_qty:
```

Comment:
and product type = service?

Reply:
I think the same logic should be followed for all products.

Take this sale order as an example:
![group_1](https://github.com/OCA/account-invoice-reporting/assets/62256279/eff4dc64-b9bc-4cda-bc15-ab2965101692)

This picking:
![group_2](https://github.com/OCA/account-invoice-reporting/assets/62256279/24a3f931-50c2-4475-ac8d-41cfe7d5e94e)

And this invoice:
![group_3](https://github.com/OCA/account-invoice-reporting/assets/62256279/1e3e3e57-86ef-48b4-80df-f1945ce22839)

This is the PDF:
![group_4](https://github.com/OCA/account-invoice-reporting/assets/62256279/32489a88-1e9e-4f7e-9bbd-13ba264ef74f)

The issue happens as well, as the amount in the invoice is less than the amount in the sale order plus the invoice line is not related to any stock.move.line as it has not been delivered.

Reply:
@carlosdauden I think you implemented this system with negatives. Do you remember why?

Reply:
@manuelregidor is right, we should consider services and consumables as well, as the same is happening in both situations.

Reply:
Well, this is considering everything (including storable ones). That's why I'm asking my colleague if he remembers about this implementation, as I think he was the one doing it.

Reply:
Yes, I said consumables as the example was with a consumable product, but the same is happening with any type of product. See this example:

Sale order with storable products:

![Captura de Pantalla 2024-03-08 a las 9 50 34](https://github.com/OCA/account-invoice-reporting/assets/32449357/5cb4862e-4b97-487a-9b1b-988d00903c4f)

Partial delivery order:

![Captura de Pantalla 2024-03-08 a las 9 50 46](https://github.com/OCA/account-invoice-reporting/assets/32449357/b9cfc930-eadf-426b-86a5-0e9a7ea26c74)

Invoicing the delivered products and just the some of the other storables (invoicing by ordered quantities):

![Captura de Pantalla 2024-03-08 a las 9 50 57](https://github.com/OCA/account-invoice-reporting/assets/32449357/7026328f-7aa4-48d0-97be-47e11b308a7e)

The report is strange like that:

![Captura de Pantalla 2024-03-08 a las 9 51 10](https://github.com/OCA/account-invoice-reporting/assets/32449357/584c3ace-dfb7-43e6-9d6c-f921f1a1636a)